tar/read.c

```diff
@@ -111,6 +111,40 @@ progress_func(void * cookie)
 	siginfo_printinfo(bsdtar, 0);
 }
 
+/*
+ * Should we skip over this file if given --resume-extract?
+ * 0: skip it.  1: don't skip.  -1: error.
+ */
+static int
+check_skip_file(const char * filename, const struct stat * archive_st)
+{
+	struct stat file_st;
+
+	/* Get info about the file on disk. */
+	if (stat(filename, &file_st) == -1) {
+		if (errno == ENOENT)
+			goto noskip;
+		goto err0;
+	}
+
+	/* Compare file size and mtime. */
```

Comment:
The comment here should point out that we're comparing the *seconds* of the mtime, not the complete timespec; some filesystems don't have sub-second timestamp precision, so comparing timespecs would produce a lot of false negatives.